### ISSUE TITLE
How do I use 2 gpus to train the same yolov3 model?

### ISSUE THREAD

**Issue by thimabru1010** [pjreddie/darknet#1848]
I'm having trouble training my yolov3 in my PC, which has 2 gpus of 8 GB. The problem is I can only train it setting Batch = 64 and Subdivisions=64. Any other value for subdivisions (32, 16, 8) generates an error: CUDA out of memory. The problem is I have 2 gpus and the program doesn't recognize the excess and allocate to the other gpu even setting flags -gpus 0,1 and compiling with CUDA. I want to lower subdivision to get a bigger accuracy. How do I use both gpus to train the same yolov3 model?

**Comment by thimabru1010** [comment]
What I want to mean: https://stackoverflow.com/questions/36313934/is-it-possible-to-split-a-network-across-multiple-gpus-in-tensorflow

Here is an example using Tensorflow in python

**Comment by AlexeyAB** [comment]
You can use 2-4x GPUs to train 2x-4x faster, so after each iteration GPUs will be synchronized.

No, you cant use lower subdivisions. Because GPU<->GPU PCI-express interconnect (~16 GB/sec) is much slower than GPU-VRAM (500 GB/sec), so training speed would be 10x-100x slower.

**Comment by thimabru1010** [comment]
But I don't want to train faster. I just want to allocate more memory.

**Comment by AlexeyAB** [comment]
Do you want to increase mini_batch size 2x and decrease performance 100x?

**Comment by thimabru1010** [comment]
Yes, because with more mini_batches I expect more accuracy during the training.

**Comment by thimabru1010** [comment]
Not decrease performance, just want more memory to train with less Subdivisions. But CUDA always runs out of memory with subs<64

**Comment by kaypositive** [comment]
> I'm having trouble training my yolov3 in my PC, which has 2 gpus of 8 GB. The problem is I can only train it setting Batch = 64 and Subdivisions=64. Any other value for subdivisions (32, 16, 8) generates an error: CUDA out of memory. The problem is I have 2 gpus and the program doesn't recognize the excess and allocate to the other gpu even setting flags -gpus 0,1 and compiling with CUDA. I want to lower subdivision to get a bigger accuracy. How do I use both gpus to train the same yolov3 model?

Please how did you resolve this issue. I am currently having same challenge

**Comment by kosson** [comment]
Same here.
